traffic_monitor/ds/stat.go

```diff
@@ -49,16 +49,16 @@ func addAvailableData(dsStats *dsdata.Stats, crStates tc.CRStates, serverCachegr
 			log.Infof("CreateStats not adding availability data for '%s': not found in Cachegroups\n", cache)
 			continue
 		}
-		deliveryServices, ok := serverDs[cache]
-		if !ok {
-			log.Infof("CreateStats not adding availability data for '%s': not found in DeliveryServices\n", cache)
-			continue
-		}
 		cacheType, ok := serverTypes[cache]
 		if !ok {
 			log.Infof("CreateStats not adding availability data for '%s': not found in Server Types\n", cache)
 			continue
 		}
+		deliveryServices, ok := serverDs[cache]
+		if !ok && cacheType != tc.CacheTypeMid {
```

Comment:
This isn't sufficient to determine that a cache is not a mid. The approach varies from place to place - TP in some places checks for names matching `.*MID.*`, queries in TO typically do something like 'MID%' - but I think the most common is to check if the type name begins with `tc.MidTypePrefix`.

Reply:
The cache types are originally from `todata.getServerTypes()` which already converts the server type to a well-defined one:

```go
		t := tc.CacheTypeFromString(serverData.Type)
```

Also, `cacheType` is of type `tc.CacheType`, so the only possible values are `CacheType("EDGE")`, `CacheType("MID")`, and `CacheType("")`

Reply:
Well thank god.